### PR TITLE
institutions: use more fields for number of papers

### PR DESCRIPTION
**Pull request by monaawi**
* INSPIR-3443
* Needs remigration

### CODE REVIEW

File: backend/inspirehep/records/api/mixins.py [hunk 1]
```diff
@@ -326,11 +328,24 @@ def get_newest_linked_conferences_uuid(self):
...
 
     def create_institution_relations(self):
-        institutions = self.get_linked_records_from_field("authors.affiliations.record")
+        institutions = list(
+            chain.from_iterable(
+                [
+                    self.get_linked_records_from_field(field)
+                    for field in self.institution_fields
+                ]
+            )
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:347]: The list is not needed here:
```suggestion
            chain.from_iterable(
                self.get_linked_records_from_field(field)
                for field in self.institution_fields
            )
```

File: backend/inspirehep/records/api/mixins.py [hunk 2]
```diff
@@ -379,17 +394,25 @@ def get_modified_institutions_uuids(self):
...
         )
-        pids_latest = self.get_linked_pids_from_field("authors.affiliations.record")
+        pids_latest = list(
+            chain.from_iterable(
+                [
+                    self.get_linked_pids_from_field(field)
+                    for field in self.institution_fields
+                ]
+            )
+        )
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:404]: same

File: backend/inspirehep/records/api/mixins.py [hunk 3]
```diff
@@ -326,11 +328,24 @@ def get_newest_linked_conferences_uuid(self):
...
+        "thesis_info.institutions.record",
+        "record_affiliations.record",
+    ]
+
     def clean_institution_literature_relations(self):
         InstitutionLiterature.query.filter_by(literature_uuid=self.id).delete()
 
     def create_institution_relations(self):
-        institutions = self.get_linked_records_from_field("authors.affiliations.record")
+        institutions = list(
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:341]: why not a `set`? also you seem to be using this several times. Wouldn't it be better to have a `linked_institutions` property on the record?

File: backend/inspirehep/records/api/mixins.py [hunk 4]
```diff
@@ -379,17 +394,25 @@ def get_modified_institutions_uuids(self):
...
-        pids_previous = set(
-            self._previous_version.get_linked_pids_from_field(
-                "authors.affiliations.record"
-            )
+        pids_previous = chain.from_iterable(
+            [
+                self._previous_version.get_linked_pids_from_field(field)
+                for field in self.institution_fields
+            ]
         )
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:414]: same

File: backend/inspirehep/records/marshmallow/literature/es.py [hunk 5]
```diff
@@ -40,7 +42,16 @@ class LiteratureElasticSearchSchema(ElasticSearchBaseSchema, LiteratureRawSchema
     _bibtex_display = fields.Method("get_bibtex_display", dump_only=True)
     abstracts = fields.Nested(AbstractSource, dump_only=True, many=True)
     author_count = fields.Method("get_author_count")
-    authors = fields.Nested(AuthorsInfoSchemaForES, dump_only=True, many=True)
+    authors = NestedWithoutEmptyObjects(
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/es.py:45]: are you sure you need this? IIRC this is expensive

File: backend/inspirehep/records/marshmallow/literature/es.py [hunk 6]
```diff
@@ -40,7 +42,16 @@ class LiteratureElasticSearchSchema(ElasticSearchBaseSchema, LiteratureRawSchema
...
+        AuthorsInfoSchemaForES,
+        default=[],
+        dump_only=True,
+        many=True,
+        attribute="authors",
+    )
+    supervisors = NestedWithoutEmptyObjects(
+        SupervisorSchema, default=[], dump_only=True, many=True, attribute="authors"
+    )
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/es.py:54]: How is this related to the rest? I thought we were already indexing the supervisors separately? or were we just removing them and not indexing them at all?

File: backend/inspirehep/records/api/mixins.py (hunk 3 above)
Comment (monaawi) [reply backend/inspirehep/records/api/mixins.py]: I needed in some place the record and in others the pids. But then I realized that we can have the same institution linked in 2 different fields in a record so I got pids for here too to apply a set and then retrieved the records.

File: backend/inspirehep/records/marshmallow/literature/es.py (hunk 6 above)
Comment (monaawi) [reply backend/inspirehep/records/marshmallow/literature/es.py]: we were just removing supervisors and I needed them indexed to look for supervisors affiliations

File: backend/inspirehep/records/api/mixins.py [hunk 7]
```diff
@@ -379,17 +380,13 @@ def get_modified_institutions_uuids(self):
         changed_deleted_status = self.get("deleted", False) ^ prev_version.get(
             "deleted", False
         )
-        pids_latest = self.get_linked_pids_from_field("authors.affiliations.record")
+        pids_latest = list(self.linked_institutions_pids)
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:383]: why a `list`...

Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:383]: are all these `list` really needed?

Comment (monaawi) [reply backend/inspirehep/records/api/mixins.py:383]: This one is needed because of ```get_records_ids_by_pids(pids_latest)``` which has ``` pids[max_batch * batch_no : max_batch * (batch_no + 1)]```
